--- a/drivers/power/reset/qcom-dload-mode.c
+++ b/drivers/power/reset/qcom-dload-mode.c
@@ -253,25 +253,6 @@ static ssize_t update_arb_store(struct kobject *kobj,
 			        struct attribute *this,
 			        const char *buf, size_t count)
 {
-	bool trigger;
-	int ret;
-
-	ret = kstrtobool(buf, &trigger);
-	if (ret < 0) {
-		pr_err("update_arb: kstrtobool failed rc=%d", ret);
-		return ret;
-	}
-
-	if (!trigger) {
-		pr_info("update_arb: trigger is false, no-op");
-		return count;
-	}
-
-	ret = qcom_scm_update_rollback_version();
-	if (ret < 0) {
-		pr_err("update_arb: SCM call TZ_UPDATE_ROLLBACK_VERSION_ID failed rc=%d", ret);
-		return ret;
-	}
 	return count;
 }
 
--- a/drivers/firmware/qcom_scm.c
+++ b/drivers/firmware/qcom_scm.c
@@ -602,43 +602,6 @@ EXPORT_SYMBOL_GPL(qcom_scm_get_download_mode);
 
 int qcom_scm_update_rollback_version(void)
 {
-	int ret;
-	struct qcom_scm_res res;
-	struct qcom_scm_desc desc = {
-		.svc = QCOM_SCM_SVC_BOOT,
-		.cmd = 0x1E,
-		.owner = ARM_SMCCC_OWNER_SIP,
-		.arginfo = QCOM_SCM_ARGS(0),
-	};
-
-	if (!__scm) {
-		pr_err("qcom_scm_update_rollback_version: SCM not initialized\n");
-		return -ENODEV;
-	}
-	memset(&res, 0, sizeof(res));
-
-	mutex_lock(&update_arb_lock);
-	if(update_arb_done == true) {
-		mutex_unlock(&update_arb_lock);
-		pr_info("qcom update arb: request already succeed\n");
-		return 0;
-	}
-	ret = qcom_scm_call(__scm->dev, &desc, &res);
-	if (ret) {
-		pr_err("qcom_scm_update_rollback_version: qcom_scm_call failed ret=%d\n", ret);
-		mutex_unlock(&update_arb_lock);
-		return ret;
-	}
-	if (res.result[0]) {
-		pr_err("qcom_scm_update_rollback_version: TZ returned status=%llu\n", res.result[0]);
-		mutex_unlock(&update_arb_lock);
-		return -ENODEV;
-	}
-
-	update_arb_done = true;
-	mutex_unlock(&update_arb_lock);
-
-	pr_info("qcom arb_update: success\n");
 	return 0;
 }
 EXPORT_SYMBOL(qcom_scm_update_rollback_version);
